lib/graphql/static_validation/literal_validator.rb

```diff
@@ -16,7 +16,7 @@ def validate(ast_value, type)
         elsif type.kind.list?
           item_type = type.of_type
           ensure_array(ast_value).all? { |val| validate(val, item_type) }
-        elsif type.kind.scalar? && !ast_value.is_a?(GraphQL::Language::Nodes::AbstractNode) && !ast_value.is_a?(Array)
+        elsif type.kind.scalar? && !ast_value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
```

Comment:
Looking at the grammar again this will mistakenly accept an enum as a scalar literal. I think I need to rework this to ensure that ast_value is recursively only composed of `GraphQL::Language::Nodes::InputObject`s and Ruby primitives.